services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/util/TimeUtils.java

```diff
@@ -0,0 +1,65 @@
+/*-
+ * ---license-start
+ * Corona-Warn-App
+ * ---
+ * Copyright (C) 2020 SAP SE and all other contributors
+ * ---
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package app.coronawarn.server.services.distribution.assembly.structure.util;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+public class TimeUtils {
+
+  private static LocalDate utcDate;
+  private static LocalDateTime utcHour;
+  private static Instant now;
```

Comment:
Is there a reason for having these as singletons that are independent of each other?
You could save a couple LoCs by:
- exposing these fields as `public`, `final` constants
- initializing them right here
- adjust the names and javadoc

What do you think?

Reply:
I was thinking that these fields can be changed by adding setters (in the future) so it can be easier to test certain components which uses them.

Reply:
Also, shouldn't we be using one LocalDate/LocalDateTime object and then converting to other formats if necessary? 

With the current solution it could happen that if `getCurrentUtcHour()` is called for the first time at 18:59 and `getUtcDate()` is called for the first time at 19:00, we have different hour values for these two properties. 

This would be solved with Michael's suggestion of keeping the static fields public because then the three values would be instantiated virtually together. But then mocking becomes a bit redundant. 

To inject values in this class on a test I had to do the following:
```java
TimeUtils.setUtcDate(LocalDate.of(1970, 1, 5));
TimeUtils.setUtcHour(LocalDateTime.of(1970, 1, 5, 0, 0));
```

Ideally, only the second line should be sufficient, and when the code call `getUtcDate()` it should return `5.1.1970`.

So my suggestion is that we keep only `LocalDateTime` property and implement `getUtcDate` something like this:
```java
public static LocalDate getUtcDate() {
  return TimeUtils.getCurrentUtcHour().toLocalDate()
}
```

Unless we have a scenario where want to inject different values?

Reply:
Done. 😄